misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt

```diff
@@ -22,23 +23,26 @@ class ConfigMetadataAction @Inject constructor(
   config: Config,
   private val jvmMetadataAction: JvmMetadataAction,
   private val mode: ConfigTabMode,
+  @ConfigRedactKey keysToRedact: List<String>
 ) : WebAction {
+  private fun redact(
+    mapOfOutputs: Map<String, String?>,
+    regex: Regex
+  ) = mapOfOutputs
+    .mapValues { it.value?.replace(regex, "████████") }
+
+  // TODO make this config variable specific redaction with an annotation on config field, not just multibound list of keys
+  // Regex to match on redacted keys for redaction in output
+  private val yamlFilesRegex = Regex("(?<=(${keysToRedact.joinToString("|")}): )([^\n]*)")
   private val resources: Map<String, String?> =
-    generateConfigResources(appName, deployment, config)
+    redact(generateConfigResources(appName, deployment, config), yamlFilesRegex)
 
   @Get("/api/config/metadata")
   @RequestContentType(MediaTypes.APPLICATION_JSON)
   @ResponseContentType(MediaTypes.APPLICATION_JSON)
   @AdminDashboardAccess
   fun getAll(): Response {
-    // TODO(mmihic): Need to figure out how to get the overrides.
```

Comment:
I'm not clear what "Need to figure out how to get the overrides" indicates here. Do we think this is still relevant?

Reply:
Good call, looks like that still is a valid TODO based on further context https://github.com/cashapp/misk/commit/7619df2889585ad85dabeefbe25f3f676153d826